cmd/populator/cmd/register/ocf_manifests.go

```diff
@@ -89,30 +113,150 @@ func runDBPopulate(ctx context.Context, src string) (err error) {
 			err = multierror.Append(err, cErr)
 		}
 	}()
-
 	session := driver.NewSession(neo4j.SessionConfig{})
 	defer func() {
 		if sErr := session.Close(); sErr != nil {
 			err = multierror.Append(err, sErr)
 		}
 	}()
 
-	var gitHash []byte
 	if cfg.UpdateOnGitCommit {
 		log.Info("APP_UPDATE_ON_GIT_COMMIT set. Updating manifests only if git commit changed.")
-		gitHash, err = getGitHash(rootDir)
+		dataInDB, err := dbpopulator.IsDataInDB(session, log, commits)
 		if err != nil {
-			return errors.Wrap(err, "while getting `git rev-parse HEAD`")
+			return errors.Wrap(err, "while verifying commits in db")
+		}
+		if dataInDB {
+			return nil
 		}
 	} else {
 		log.Info("APP_UPDATE_ON_GIT_COMMIT not set. Ignoring git commit, always updating manifests.")
 	}
 
+	for _, src := range sourcesInfo {
+		err = runDBPopulate(ctx, cfg, session, log, src)
+		if err != nil {
+			return errors.Wrap(err, "while populating db")
+		}
+	}
```

Comment:
It cannot be in this layer you need to push that much lower. 

You have two options:
- merge all downloaded folder to a single folder and pass single source (merged folder) to populator
- pass all sources to populator and `populate` function aware of that, so you collect kinds from all sources and populate them. 


----
Currently, you execute populator main logic for each sources and only the last wins, rest are removed, check this function body:
https://github.com/capactio/capact/blob/21d6301e6ab628c089e1b13c35a74f0360a7b985/pkg/sdk/dbpopulator/populator.go#L464-L479

Reply:
Corrected - I decided to use the second approach as it seems to be more straightforward and more maintainable.